content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx

```diff
@@ -0,0 +1,530 @@
+- `updateMany`: Update many composite types at once
+- `deleteMany`: Delete many composite types at once
+
+For example, use `push` to add a new photo to the `photos` list:
+
+```ts
+const product = prisma.product.update({
+  where: {
+    id: 10,
+  },
+  data: {
+    photos: {
+      push: [{ height: 100, width: 200, url: '1.jpg' }],
+    },
+  },
+})
+```
+
+Use `updateMany` to update photos with a `url` of `1.jpg` or `2.png`:
+
+```ts
+const product = prisma.product.update({
+  where: {
+    id: 10,
+  },
+  data: {
+    photos: {
+      updateMany: {
+        where: {
+          url: '1.jpg',
+        },
+        data: {
+          url: '2.png',
+        },
+      },
+    },
+  },
+})
+```
+
+The following example uses `deleteMany` to delete all photos with a `height` of 100:
+
+```ts
+const product = prisma.product.update({
+  where: {
+    id: 10,
+  },
+  data: {
+    photos: {
+      deleteMany: {
+        where: {
+          height: 100,
+        },
+      },
+    },
+  },
+})
+```
+
+## Deleting composite types with <inlinecode>delete</inlinecode> and <inlinecode>deleteMany</inlinecode>
+
+To remove records which embed a composite type, use the `delete` or `deleteMany` methods. This will also remove the embedded composite type.
+
+For example, use `deleteMany` to delete all products with a `size` of `"Small"`. This will also delete any embedded `photos`.
+
+```ts
+const deleteProduct = await prisma.product.deleteMany({
+  where: {
+    size: 'Small',
+  },
+})
+```
+
+## Filtering composite types with <inlinecode>find</inlinecode> and <inlinecode>findMany</inlinecode>
+
+Composite types can be filtered within a `find` or `findMany` method. The following section describes the methods available for filtering down to a single type or multiple types, and gives examples of each.
+
+### Filter for one composite type
+
+Use the new `is`, `equals` and `isNot` methods to change a single composite type:
+
+- `is`: Filter results by matching composite types. Requires one or more fields to be present _(e.g. Filter orders by the street name on the shipping address)_
+- `equals`: Filter results by matching composite types. Requires all fields to be present. _(e.g. Filter orders by the full shipping address)_
+- `isNot`: Filter results by non-matching composite types.
+
+For example, use `is` to filter for orders with a street name of `'555 Candy Cane Lane'`:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      is: {
+        street: '555 Candy Cane Lane',
+      },
+    },
+  },
+})
+```
+
+You can also use a shorthand notation for this query, where you leave out the `is` and specify just the fields that you want to filter for:
+
+```ts
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      street: '555 Candy Cane Lane',
+    },
+  },
+})
+```
+
+Use `equals` to filter for orders which match on all fields in the shipping address:
```

Comment:
Maybe it's just me, but I always get confused about the word filtering. Does filtering mean: excluding or including matching results?

I'd prefer we use the word "select".

Use `equals` to select orders which match...

Reply:
I can change this to `select`... do you also have a suggestion for the comment above where you say 'Hmm, what do you mean?' (sorry can't see how to reply to this directly)

Reply:
Hahh... I think a prior comment got deleted. Also just deleted the "Hmm, what do you mean".